File: Leetcode/determine_if_string_halves_are_alike/halves_alike.rb
# @param {String} s
# @return {Boolean}
def halves_are_alike(s)
  half = s.length / 2
  s[0..half-1].gsub(/[^aeiouAEIOU]/, '').length == s[half..-1].gsub(/[^aeiouAEIOU]/, '').length
end

p halves_are_alike("book") == true
p halves_are_alike("textbook") == false

File: Leetcode/determine_if_string_halves_are_alike/halves_alike.py
def halves_are_alike(s):
    half = len(s) // 2
    return len([char for char in s[:half] if char.lower() in 'aeiou']) == len([char for char in s[half:] if char.lower() in 'aeiou'])

print(halves_are_alike("book") == True)
print(halves_are_alike("textbook") == False)